python/mxnet/optimizer.py

```diff
@@ -80,6 +71,20 @@ def set_lr_scale(self, args_lrscale):
         """
         self.lr_scale = args_lrscale.copy()
 
+    def _update_count(self, index):
+        """
+        update num_update
+
+        Parameters:
+        index : int
+            The index will be updated
+        """
+        if index not in self._index_update_count:
+            self._index_update_count[index] = 0
+        self._index_update_count[index] += 1
+        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
Ok, i think your `num_updates` are the same as my `version` to some extent. I just feel like having different learning rates per workers are strange. BTW, how momentum is done in ASGD settings right now? Does each worker keep their own momentum of weights?